### PR TITLE
zabbix_host : add the ability to set inventory_mode #1005

### PR DESCRIPTION
**Pull request by ritzk**
Allows user to set inventory_mode . Had to move this to a different branch. Closed the old request, and opened a new one.

### CODE REVIEW

File: monitoring/zabbix_host.py [hunk 1]
```diff
@@ -61,6 +61,11 @@
             - List of templates linked to the host.
         required: false
         default: None
+    inventory_mode:
+        description: Confiugrate the inventory mode
+        choices: ['automatic', 'manual', 'disabled' ]
+        required: false
+        default: None
```
Comment (bcoca) [inline monitoring/zabbix_host.py:69]: needs version added

**Comment by ritzk** [comment]
done

**Comment by robynbergeron** [comment]
Hi @ritzk -- thanks for the PR.

FYI to all that this is a redo of #1005 (which is now closed), in case you previously looked at that one.

@cove @harrisongu - Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: monitoring/zabbix_host.py [hunk 2]
```diff
@@ -356,6 +363,28 @@ def link_or_clear_template(self, host_id, template_id_list):
...
+    def update_inventory_mode(self, host_id, inventory_mode):
+
+	if not inventory_mode:
+            return
+
+        if inventory_mode == "automatic":
+            inventory_mode = int(1)
+        elif inventory_mode == "manual":
+            inventory_mode = int(0)
+        else: # inventory_mode == "disabled":
```
Comment (resmo) [inline monitoring/zabbix_host.py]: when I read the code the default seems to be `disabled` but the doc says 'none'?

Comment (ritzk) [inline monitoring/zabbix_host.py]: -    def update_inventory_mode(self, host_id, inventory_mode): 
  ...
-   if not inventory_mode:
-            return
  ...
  def main(): 
  ...
  -            inventory_mode=dict(required=False, choices=["automatic", "manual", "disabled"]), 
    ...
  -    inventory_mode = module.params['inventory_mode'] 

The default is None, as inventory_mode is not set. This function would be executed.

zabbix server translates this as disabled by default, if a value is not set.

**Comment by gregdek** [comment]
@cove @harrisongu This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by cove** [comment]
+1 after the spelling correction

**Comment by ritzk** [comment]
@cove fixed.

**Comment by eikef** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @ritzk. @eikef please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Thanks again to @ritzk for this PR, and thanks @eikef for reviewing. Marking for inclusion.

**Comment by gregdek** [comment]
Oops... looks like the bot jumped the gun.

@ritzk Thanks for submitting this pull request. Unfortunately, it appears that the underlying code has changed since your submission, and this PR now needs to be rebased. 

If you believe that this PR still applies, please rebase and we will put it into our community review process as soon as possible. 

Once you have rebased this PR, please add a comment with the text "ready_for_review" and we will continue with the review process.  Thanks!

**Comment by ritzk** [comment]
waiting on https://github.com/ansible/ansible-modules-extras/pull/1850

**Comment by ritzk** [comment]
ready_for_review . fixed .

**Comment by eikef** [comment]
shipit

**Comment by resmo** [comment]
Thanks again to @ritzk for this PR, and thanks @eikef for reviewing. Marking for inclusion.

**Pull request merged**